### PR TITLE
add event_time_microseconds field to system.part_log table

### PR DESCRIPTION
**Pull request by bharatnc**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

New `event_time_microseconds column` in `system.part_log` table.

Detailed description / Documentation draft:

This PR adds a new column called `event_time_microseconds` which adds microseconds precision to the already existing `event_time` column.

```sql
SELECT *
FROM system.part_log
LIMIT 1
FORMAT Vertical

Row 1:
───────
query_id:                8747499d-c6e4-4027-9a1b-580c5af6a6d7
event_type:              NewPart
event_date:              2021-02-03
event_time:              2021-02-03 00:27:09
event_time_microseconds: 2021-02-03 00:27:09.155966
duration_ms:             1249
database:                default
table:                   table_with_single_pk
part_name:               all_126_126_0
partition_id:            all
path_on_disk:            ./store/ff3/ff345ee0-a15f-467f-8c9f-dcc5448cd3bf/all_126_126_0/
rows:                    1048545
size_in_bytes:           23922
merged_from:             []
bytes_uncompressed:      0
read_rows:               0
read_bytes:              0
peak_memory_usage:       0
error:                   0
exception:               
       

85 rows in set. Elapsed: 0.056 sec. 
```

Related  to: #9800

### CODE REVIEW

**Comment by bharatnc** [comment]
@tavplubix would you mind reviewing ? This is one last PR (system.part_log was left to add `event_time_microseconds` field) and is similar to: https://github.com/ClickHouse/ClickHouse/pull/14760 that you had reviewed in the past. Thank you!

File: tests/queries/0_stateless/01686_event_time_microseconds_part_log.sql [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            LIMIT 1
+        ) AS time_with_microseconds,
+        (
+            SELECT event_time
+            FROM system.part_log
+            ORDER BY event_time DESC
+            LIMIT 1
+        ) AS time
+    )
+SELECT if(dateDiff('second', toDateTime(time_with_microseconds), toDateTime(time)) = 0, 'ok', 'fail');
```
Comment (tavplubix) [inline tests/queries/0_stateless/01686_event_time_microseconds_part_log.sql:29]:
```suggestion
            SELECT (event_time, event_time_microseconds)
            FROM system.part_log
            ORDER BY event_time DESC
            LIMIT 1
        ) AS time
    )
SELECT if(dateDiff('second', toDateTime(time.2), toDateTime(time.1)) = 0, 'ok', 'fail');
```

Comment (tavplubix) [reply tests/queries/0_stateless/01686_event_time_microseconds_part_log.sql:29]: It should fix Functional stateless tests flaky check failure

**Review by tavplubix — APPROVED** [review]

File: tests/queries/0_stateless/01686_event_time_microseconds_part_log.sql (hunk 1 above)
Comment (bharatnc) [reply tests/queries/0_stateless/01686_event_time_microseconds_part_log.sql]: Thank you.

**Comment by bharatnc** [comment]
Rebased to fix merge conflicts. Also added `01686_event_time_microseconds_part_log` to arcadia skip list similar to other tests involving `event_time_microseconds`.

**Comment by tavplubix** [comment]
AST fuzzer (UBSan) - #20232
Stress test (thread) - #20032
Yandex third-party checks - looks like an issue in CI infrastructure

**Pull request merged**